File: Alg1_2.cpp
#include<iostream>
#include<algorithm>

using std::cout;
using std::cin;
using std::string;
using std::max;


int main()
{
    int REPLACE;
    string MYSTR;
    cin >> REPLACE;
    cin >> MYSTR;
    int lenght = size(MYSTR);
    int maximum = 0;



    for(int symbol = 97; symbol < 123; symbol++)
    {
        int j = 0;
        for(int i = 0; i < lenght; i++)
        {
            while (REPLACE >= 0 && j != lenght)
            {
                if (MYSTR[j] != (char)symbol and REPLACE != 0)
                {
                    REPLACE--;
                    j++;
                }
                else if (MYSTR[j] != (char)symbol && REPLACE == 0)
                {
                    break;
                }
                else
                {
                    j++;
                }
            }
            maximum = max(j - i, maximum);
            if (MYSTR[i] != (char)symbol)
            {
                REPLACE++;
            }
            
        }
    }

    cout << maximum;

    

    return 0;
}

File: Alg1_2.py
REPLACE = int(input())
MYSTR = input()
lenght = len(MYSTR)
maximum = 0

for symbol in range(97, 123):
        j = 0
        symbol = chr(symbol)
        
        for i in range(lenght):
            while REPLACE >= 0 and j != lenght:
                if MYSTR[j] != symbol and REPLACE != 0:
                    REPLACE -= 1
                    j += 1
                elif MYSTR[j] != symbol and REPLACE == 0:
                    break
                else:
                    j += 1
            maximum = max(len(MYSTR[i:j]), maximum)
            if MYSTR[i] != symbol:
                REPLACE += 1

print(maximum)